include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,234 @@
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a any specification in a given subset
+  //====================================================================================================================
+  template<typename O, typename... Specs>
+  concept is_supported = (supports_option<Specs,O> || ... );
+}
+
+namespace eve::option
+{
+  //====================================================================================================================
+  // Internal option carrying conditional mask or conditional expressions
+  //====================================================================================================================
+  struct condition_ : rbr::as_keyword<condition_>
+  {
+    using rbr::as_keyword<condition_>::operator=;
+    template<typename T> static constexpr bool check()
+    {
+      return conditional_expr<std::remove_cvref_t<T>> || logical_value<std::remove_cvref_t<T>>;
+    }
+  };
+
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @var condition
+  //!   @brief Keyword for retrieving conditionals decorator
+  //!
+  //!   When a eve::decorated_callable is called with an option provided vi eve::conditional, the value of the
+  //!   conditional can be retrieved using eve::condition
+  //!   @see eve::supports
+  //! @}
+  //====================================================================================================================
+  inline constexpr condition_ condition = {};
+}
+
+namespace eve::detail
+{
+  //====================================================================================================================
+  // Intermediate type for handling chains of option like in func[opt1 = y][opt2 = x][ignore](...)
+  //====================================================================================================================
+  template<typename Tag, typename Settings, typename... Specs> struct decorated_fn
+  {
+    using callable_tag_type = void;
+
+    template<is_supported<Specs...> O>
+    EVE_FORCEINLINE auto operator[](O const& o) const
+    {
+      if constexpr(rbr::concepts::option<O>)
+      {
+        // Merge new option with current ones
+        auto new_opts = options{rbr::merge(rbr::settings{o}, opts)};
+        return decorated_fn<Tag,decltype(new_opts),Specs...>{new_opts};
+      }
+      // Wrap conditionals and booleans into a condition option
+      else if constexpr(conditional_expr<O> || eve::logical_value<O>)
+      {
+        return (*this)[option::condition = o];
+      }
+      else if constexpr(std::same_as<O,bool>)
+      {
+        using type = std::conditional_t<std::same_as<bool,O>,std::uint8_t,O>;
+        return (*this)[option::condition = if_(logical<type>(o))];
+      }
+    }
```

Comment:
If I undersand this correctly, I don't like it. 

tbh we never used `opt =` syntax, we always used tagged types.

I'd expect each "supports" to react with it's own option and update the common set of options.
Conditional should not be special

Reply:
We used tagged type except for a bunch of place where the option is indeed a single value defined as a `some_kw = the value`. See unroll for example.

I need to wrap the logical/conditional into a rbr keyword, that's the API opf RBR. I don't think addign all the bagage of rbr::keyword to logical and all other thing is a good idea.